src/sql/workbench/contrib/query/browser/queryActions.ts

```diff
@@ -791,6 +791,11 @@ export class ListDatabasesActionItem extends Disposable implements IActionViewIt
 	}
 
 	private updateConnection(databaseName: string): void {
+		// Ignore if the database name is not provided, this happens when the query editor connection is changed to
+		// a provider that does not support database.
+		if (!databaseName) {
+			return;
+		}
 		this._isConnected = true;
```

Comment:
Wouldn't we still want to set this?

Reply:
I don't think it matters, because it is not visible anyways. when we change the connection to SQL Server, the code path will be executed again and all the states will be correctly set.

Reply:
Ah, didn't see that this was only within the context of the list databases action. (Would have been clearer to just name it enabled, but whatever)